client/src/components/board-details/board-information/BoardNameField.tsx

```diff
@@ -0,0 +1,86 @@
+"use client";
+import React, { KeyboardEvent, useEffect } from "react";
+import { Pencil, Save, XCircle } from "lucide-react";
+import { Button } from "@/components/ui/button";
+import BoardInfoInputItem from "@/components/board-details/board-information/BoardInfoInputItem";
+import { useBoardName } from "@/hooks/board-details/useBoardName";
+import { BoardResponse } from "@/interfaces/responses/user-boards/board-response";
+
+interface BoardNameFieldProps {
+  board: BoardResponse | undefined;
+  id: string;
+  mutate: () => void;
+}
+
+const BoardNameField: React.FC<BoardNameFieldProps> = ({ board, id, mutate }) => {
+  const { name, isEditing, updating, setName, handleEditClick, handleCancel, handleConfirm } = useBoardName(
+    board,
+    mutate
+  );
+
+  const handleKeyDown = (e: KeyboardEvent<HTMLInputElement>) => {
+    if (e.key === "Enter") {
+      handleConfirm();
+    } else if (e.key === "Escape") {
+      handleCancel();
+    }
+  };
+
+  useEffect(() => {
+    if (board) {
+      setName(board.name);
+    }
+  }, [board]);
```

Comment:
I did this on purpose. If you try to add `setName` there, it will cause an infinite loop :PPP. Sometimes, not everything should be added to the dependency array, even if the linter suggests otherwise ;).

Reply:
Oh, I haven't noticed it :)